test/test_integration.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
```

Comment:
The ideas of tests are good but you should indeed write them to to be run by `pytest`.